pkg/account/service.go

```diff
@@ -133,6 +138,68 @@ func (a *Service) Update(ID string, data *Account) (*Account, error) {
 	return account, nil
 }
 
+// Create creates a new account using the data provided. Returns the account record
+func (a *Service) Create(data *Account) (*Account, error) {
+	// Validate the incoming record doesn't have unneeded fields
+	err := validation.ValidateStruct(data,
+		// This may be considered double validation but we are going to need the ID
+		// so need to make sure its set
+		validation.Field(&data.ID, validateID...),
+		validation.Field(&data.AdminRoleArn, validateAdminRoleArn...),
+		validation.Field(&data.LastModifiedOn, validation.By(isNil)),
+		validation.Field(&data.Status, validation.By(isNil)),
+		validation.Field(&data.CreatedOn, validation.By(isNil)),
+		validation.Field(&data.PrincipalRoleArn, validation.By(isNil)),
+		validation.Field(&data.PrincipalPolicyHash, validation.By(isNil)),
+	)
+	if err != nil {
+		return nil, errors.NewValidation("account", err)
+	}
+
+	// Check if account already exists
+	existingAccount, err := a.Get(*data.ID)
+	if existingAccount != nil {
+		return nil, errors.NewAlreadyExists("account", *data.ID)
+	}
+	if err != nil {
+		if !errors.Is(err, errors.NewNotFound("account", *data.ID)) {
+			return nil, err
+		}
+	}
+
+	new, err := NewAccount(NewAccountInput{
+		ID:                *data.ID,
+		AdminRoleArn:      *data.AdminRoleArn,
+		Metadata:          data.Metadata,
+		PrincipalRoleName: a.principalRoleName,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.managerSvc.MergePrincipalAccess(new)
```

Comment:
Should we update this to UpsertPrincipalAccess?

Reply:
Done